### PR TITLE
Lucy readme

### PR DESCRIPTION
**Pull request by QuiteStochastic**
significant improvements to the readme for Lucy, formerly know as Gauntlet, formerly known as "the integration test framework"

### CODE REVIEW

File: src/app/test_executive/README.md [hunk 1]
```diff
@@ -1,120 +1,250 @@
...
+
+Given this, the solution, therefore, is quite clear: we must spin up from scratch an entire decentralized network of nodes, and then run tests against this network.  In concept this is quite simple, but of course in implementation it is ambitious.
+
+#### Structure (super high level)
+
+There are a number of moving parts to gauntlet, and it is also actively being developed and extended.  However in it's current state, Gauntlet tests Mina Protocol by creating a whole entire Mina testnet from scratch with a specified number of nodes of various types, creating a new blockchain from genesis, and then monitoring and interacting with this testnet in accordance to test logic as determined in pre-written tests; tests which are written in an Ocaml DSL which we have developed.  If the testnet seems healthy and behaves as one expects in response to the interactions, then the test passes, if not it fails.  Then the testnet is destroyed.  If one wants, one can prevent the destruction of the testnet with a flag, and then manually interact with the nodes on the testnet.
```
Comment (rbonichon) [inline src/app/test_executive/README.md]: Ocaml -> OCaml
```suggestion
There are a number of moving parts to gauntlet, and it is also actively being developed and extended.  However in it's current state, Gauntlet tests Mina Protocol by creating a whole entire Mina testnet from scratch with a specified number of nodes of various types, creating a new blockchain from genesis, and then monitoring and interacting with this testnet in accordance to test logic as determined in pre-written tests; tests which are written in an OCaml DSL which we have developed.  If the testnet seems healthy and behaves as one expects in response to the interactions, then the test passes, if not it fails.  Then the testnet is destroyed.  If one wants, one can prevent the destruction of the testnet with a flag, and then manually interact with the nodes on the testnet.
```

File: src/app/test_executive/README.md [hunk 2]
```diff
@@ -1,120 +1,250 @@
...
+
+#### Tests
+
+The usecase which Gauntlet has been designed around (thus far) is the usecase of running automated *tests*.  There are a number of pre-written tests which are compiled into the test_executive (such as the payments_test, zkapps_test, chain_reliability_test, etc).  Those who are familiar with Ocaml can learn the DSL and modify or extend these tests, and/or write new tests.
```
Comment (rbonichon) [inline src/app/test_executive/README.md]:
```suggestion
The usecase which Gauntlet has been designed around (thus far) is the usecase of running automated *tests*.  There are a number of pre-written tests which are compiled into the test_executive (such as the payments_test, zkapps_test, chain_reliability_test, etc).  Those who are familiar with OCaml can learn the DSL and modify or extend these tests, and/or write new tests.
```

File: src/app/test_executive/README.md [hunk 3]
```diff
@@ -1,120 +1,250 @@
...
+
+The env vars `GOOGLE_APPLICATION_CREDENTIALS`, `GCLOUD_API_KEY`, `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_DEFAULT_REGION` are the same env vars that you set up in the earlier section of this readme [ GCP credentials and infrastructure related env vars](# GCP credentials and infrastructure related env vars).  The `--mount` argument tells docker to put the keyfile of the service account onto the file system inside of the container, so that processes running inside the container can access the keyfile.  The env var `GOOGLE_APPLICATION_CREDENTIALS` is to be set to the path to the keyfile inside the container, not the path to the keyfile on the host machine (if you don't mess with other parts of the idiomatic expression, then you don't need to modify this `--env` flag).
+
+
+#### Compile Gauntlet from source
+
+If you wish to modify or extend existing tests, and/or write a whole new test, or modify the Gauntlet test executive itself, you will need to compile the Gauntlet test_executive from source.  Gauntlet is a complex piece of software written in Ocaml, it's not some simple python or bash script.
```
Comment (rbonichon) [inline src/app/test_executive/README.md]:
```suggestion
If you wish to modify or extend existing tests, and/or write a whole new test, or modify the Gauntlet test executive itself, you will need to compile the Gauntlet test_executive from source.  Gauntlet is a complex piece of software written in OCaml, it's not some simple python or bash script.  
```

File: src/app/test_executive/README.md [hunk 4]
```diff
@@ -1,120 +1,250 @@
...
+
+
+#### Compile Gauntlet from source
+
+If you wish to modify or extend existing tests, and/or write a whole new test, or modify the Gauntlet test executive itself, you will need to compile the Gauntlet test_executive from source.  Gauntlet is a complex piece of software written in Ocaml, it's not some simple python or bash script.  
+
+Gauntlet is in the same git repository as the rest of the Mina Daemon at https://github.com/MinaProtocol/mina.  
+
+I will assume the user who wishes to compile Gauntlet is familiar with not just Ocaml but also with the normal compilation process of mina.  Compiling the test executive is not that different.
```
Comment (rbonichon) [inline src/app/test_executive/README.md]:
```suggestion
I will assume the user who wishes to compile Gauntlet is familiar with not just OCaml but also with the normal compilation process of mina.  Compiling the test executive is not that different.
```

File: src/app/test_executive/README.md [hunk 5]
```diff
@@ -1,120 +1,250 @@
...
-# Code Structure
+## Code Structure
 
-### Integration Test general purpose directories
+### Gauntlet general purpose directories
 
-- `src/app/test_executive/` — The integration tests themselves live here, along with the `test_executive.exe` entrypoint for executing them.
-- `src/lib/integration_test_lib/` — Contains the core logic for integration test framework. This is where you will find the implementation of the DSL, including the event router, network state, and wait conditions. This library also contains the definition of the interfaces for execution engines and test definitions.
+- `src/app/test_executive/` — The pre-written Gauntlet tests live here, along with the `test_executive.ml` which is the entrypoint for executing them.
+- `src/lib/integration_test_lib/` — Contains the core logic for integration test framework. This is where you will find the implementation of the Gauntlet Ocaml DSL, the event router, the network state data structure, and wait conditions. This library also contains the definition of the interfaces for execution engines and test definitions.
```
Comment (rbonichon) [inline src/app/test_executive/README.md]:
```suggestion
- `src/lib/integration_test_lib/` — Contains the core logic for integration test framework. This is where you will find the implementation of the Gauntlet OCaml DSL, the event router, the network state data structure, and wait conditions. This library also contains the definition of the interfaces for execution engines and test definitions.
```

File: src/app/test_executive/README.md [hunk 6]
```diff
@@ -1,120 +1,250 @@
...
-- `automation/terraform/testnets` — During runtime, when using the GCP cloud engine, a directory of the same name as your testnet name will be created within this directory.  the terraform file which is `terraform apply`'d (ie main.tf.json) lives in here.
-- `automation/terraform/modules/o1-integration` and `automation/terraform/modules/o1-testnet` — many terraform modules which are referenced by main.tf.json will be found in these directories
-- `helm` — The helm charts (detailed yaml files) which fully specifies the configuration of all the nodes in GCP live here
+- `src/lib/integration_test_cloud_engine/` — This library is the current implementation of the GCP cloud based execution engine, which deploys testnets in Gcloud's GKE environment.  As with any engine, it implements the interface defined in `Integration_test_lib`.  This execution engine leverages a good deal of our existing coda automation system.  These files are Ocaml.
```
Comment (rbonichon) [inline src/app/test_executive/README.md:254]:
```suggestion
- `src/lib/integration_test_cloud_engine/` — This library is the current implementation of the GCP cloud based execution engine, which deploys testnets in Gcloud's GKE environment.  As with any engine, it implements the interface defined in `Integration_test_lib`.  This execution engine leverages a good deal of our existing coda automation system.  These files are OCaml.
```

**Review by rbonichon — COMMENTED** [review]
Nit: Ocaml is actually written OCaml

File: src/app/test_executive/README.md (hunk 1 above)
Comment (QuiteStochastic) [reply src/app/test_executive/README.md]: thanks for the suggestions, have applied this change manually.

**Pull request merged**